core/message-handling.go

```diff
@@ -38,33 +39,30 @@ type messageStreamHandler func(in <-chan []byte, reply chan<- []byte)
 // message hasn't been processed before.
 type messageHandler func(msg interface{}) (reply <-chan interface{}, new bool, err error)
 
-// generatedUIMessageHandler receives generated messages from in
-// stream, assigns a UI to it and arranges it to be delivered to peer
-// replicas.
-type generatedUIMessageHandler func(in <-chan messages.MessageWithUI)
+// generatedUIMessageHandler assigns and attaches a UI to a generated
+// message and arranges it to be delivered to peer replicas. It is
+// safe to invoke concurrently.
+type generatedUIMessageHandler func(msg messages.MessageWithUI)
 
-// uiMessageConsumer receives generated messages with UI attached and
-// arranges them to be delivered to peer replicas.
+// uiMessageConsumer receives a generated message with UI attached and
+// arranges it to be delivered to peer replicas.
 type uiMessageConsumer func(msg messages.MessageWithUI)
 
 // defaultMessageHandler construct a standard messageHandler using id
 // as the current replica ID and the supplied interfaces.
 func defaultMessageHandler(id uint32, log messagelog.MessageLog, config api.Configer, stack Stack) messageHandler {
 	n := config.N()
-	uiMessageChan := make(chan messages.MessageWithUI)
 
 	view := func() uint64 { return 0 } // view change is not implemented
 	clientStates := clientstate.NewProvider()
 	acceptUI := defaultUIAcceptor(stack)
 	collectCommit := defaultCommitCollector(id, clientStates, config, stack)
-	handleGeneratedUIMessages := defaultGeneratedUIMessageHandler(stack, log)
+	handleGeneratedUIMessage := defaultGeneratedUIMessageHandler(stack, log)
```

Comment:
> And we can see the term "generated UI message" many times, does the word "generated" have special meaning? I mean that if you intend to use a separate counterpart like "non-generated" UI message somewhere, that's fine. But otherwise I thought we can go with only "UIMessage (meaning message with UI)" and "message (meaning message without UI)" .

The intention was to refer to a generated message that should have a UI assigned ("message with UI"). "Generated" was supposed to stress that the message has been just generated and has no UI assigned yet. 

As of identifier names in the code, `UIMessage` was a concise way to refer to "a message with UI (to be) assigned". Do you think it would make sense to change `GeneratedUIMessageHandler` and friends to something like `GeneratedMessageWithUIHandler`?